# Task 1: Data preparation

In [3]:
# import packages

import pandas as pd

c:\Users\hetpe\anaconda3\Lib\site-packages\pandas\core\computation\expressions.py:22: UserWarning: Pandas requires version '2.10.2' or newer of 'numexpr' (version '2.10.1' currently installed).
  from pandas.core.computation.check import NUMEXPR_INSTALLED


In [7]:
df_mood = pd.read_csv('dataset_mood_smartphone.csv')
df_mood

,Unnamed: 0,id,time,variable,value
0,1,AS14.01,2014-02-26 13:00:00.000,mood,6.000
1,2,AS14.01,2014-02-26 15:00:00.000,mood,6.000
2,3,AS14.01,2014-02-26 18:00:00.000,mood,6.000
3,4,AS14.01,2014-02-26 21:00:00.000,mood,7.000
4,5,AS14.01,2014-02-27 09:00:00.000,mood,6.000
...,...,...,...,...,...
376907,2770399,AS14.30,2014-04-11 07:51:16.948,appCat.weather,8.032
376908,2772465,AS14.30,2014-04-19 11:00:32.747,appCat.weather,3.008
376909,2774026,AS14.30,2014-04-26 10:19:07.434,appCat.weather,7.026
376910,2774133,AS14.30,2014-04-27 00:44:48.450,appCat.weather,23.033


In [ ]:
print(df_mood.id.unique())

time_measures = []
for id in df_mood.id.unique():
    id_df = df_mood[df_mood.id == id]

    time_measures.append(len(id_df.time.unique()))

print(sorted(time_measures))

<ArrowStringArray>
['AS14.01', 'AS14.02', 'AS14.03', 'AS14.05', 'AS14.06', 'AS14.07', 'AS14.08',
 'AS14.09', 'AS14.12', 'AS14.13', 'AS14.14', 'AS14.15', 'AS14.16', 'AS14.17',
 'AS14.19', 'AS14.20', 'AS14.23', 'AS14.24', 'AS14.25', 'AS14.26', 'AS14.27',
 'AS14.28', 'AS14.29', 'AS14.30', 'AS14.31', 'AS14.32', 'AS14.33']
Length: 27, dtype: str
[2270, 3096, 3429, 6829, 8743, 10321, 10689, 10795, 11055, 12188, 13657, 13738, 13939, 14100, 15023, 15133, 15474, 15490, 15777, 16434, 16716, 16809, 17334, 18720, 18884, 20985, 21298]


There are 276911 instances (without headers) in the raw data and 5 colomns
- 27 participants

Amount of measures in time differs for the participants. Ranging from 2270 times, to 21298 times

